### PR TITLE
Remove everything but LiveOS/ from appended ISO

### PR DESCRIPTION
**Pull request by lzap**
I work on Foreman Discovery project which aims to create hardware discovery
livecd based on Fedora/CentOS/RHEL. We use the PXE script from this package but
because size matters, I found a way to trim down the resulting init RAM disk a
bit.

It looks like the only required file on the appended ISO image is
LiveOS/squashfs.img and all the others are unnecessary. It looks like in our
case (220 MB source ISO) this is about 33 MB of PXELinux/EFI loader code and
configuration, which is a decent amount to drop.

This patch re-creates the ISO without those files before it is appended to the
init RAM disk. It introduces new dependency: mkisofs, so I will likely need to
add a RPM dependency. Overall, there should be no functionality changes, but
the resulting image is smaller.

### CODE REVIEW

**Comment by lzap** [comment]
@fabiand - definitive solution to oversized macboot.efi? ;-)

**Comment by vpodzime** [comment]
Looks good to me except that the dependency should be added.

File: tools/livecd-iso-to-pxeboot.sh [hunk 1]
```diff
@@ -27,10 +27,11 @@ usage() {
 }
 
 cleanup() {
+    [ -d "$STRIPPEDISO" ] && rm -rf $STRIPPEDISO
     [ -d "$CDMNT" ] && umount $CDMNT && rmdir $CDMNT
 }
 
-exitclean() {
+cleanup_error() {
```
Comment (lzap) [inline tools/livecd-iso-to-pxeboot.sh]: For the record, I renamed this and improved the exit hooks.

File: tools/livecd-iso-to-pxeboot.sh [hunk 2]
```diff
@@ -67,16 +68,17 @@ if [ -d tftpboot ]; then
...
-# FIXME: would be better if we had better mountpoints
-CDMNT=$(mktemp -d /media/cdtmp.XXXXXX)
-mount -o loop "$ISO" $CDMNT || exitclean
+CDMNT=$(mktemp -d /tmp/$0-mount.XXXXXX)
+STRIPPEDISO=$(mktemp -d /tmp/$0-stripped.XXXXXX)
+mount -o loop "$ISO" $CDMNT || cleanup_error
 
-trap exitclean SIGINT SIGTERM
+trap cleanup_error SIGINT SIGTERM
+trap cleanup EXIT
```
Comment (lzap) [inline tools/livecd-iso-to-pxeboot.sh]: Now there are two of them, with/wo error (return code 0/1).

File: tools/livecd-iso-to-pxeboot.sh [hunk 3]
```diff
@@ -67,16 +68,17 @@ if [ -d tftpboot ]; then
 fi
 
 # Mount the ISO.
-# FIXME: would be better if we had better mountpoints
-CDMNT=$(mktemp -d /media/cdtmp.XXXXXX)
-mount -o loop "$ISO" $CDMNT || exitclean
+CDMNT=$(mktemp -d /tmp/$0-mount.XXXXXX)
+STRIPPEDISO=$(mktemp -d /tmp/$0-stripped.XXXXXX)
```
Comment (lzap) [inline tools/livecd-iso-to-pxeboot.sh]: Also this essentially moves the temp from `/media` to `/tmp` which is usually tmpfs on modern Fedoras. I believe if people have anough swap, this should not be a problem?

File: tools/livecd-iso-to-pxeboot.sh [hunk 4]
```diff
@@ -93,9 +95,11 @@ mkdir tftpboot
 # initrd image.  The Linux kernel will do the right thing,
 # aggregating both cpio archives (initrd + ISO) into a single
 # filesystem.
-ISOBASENAME=`basename "$ISO"`
-ISODIRNAME=`dirname "$ISO"`
-( cd "$ISODIRNAME" && echo "$ISOBASENAME" | cpio -H newc --quiet -L -o ) |
+NEWISO=$STRIPPEDISO/`basename "$ISO"`
+mkisofs --quiet -J -joliet-long -r -T -o $NEWISO --root LiveOS $CDMNT/LiveOS/squashfs.img
```
Comment (lzap) [inline tools/livecd-iso-to-pxeboot.sh]: And here is the thing, this creates simple ISO with joliet (I guess that's not even necessary).

**Comment by lzap** [comment]
Okay, how to proceed with the SPEC change? Where do I find the upstream SPEC? Is it only in the Fedora dist-git?

File: tools/livecd-iso-to-pxeboot.sh (hunk 3 above)
Comment (bcl) [inline tools/livecd-iso-to-pxeboot.sh]: > I believe if people have anough swap, this should not be a problem?

I'd rather this went to /var/tmp/ given the size an iso can be.

**Comment by bcl** [comment]
> Okay, how to proceed with the SPEC change? Where do I find the upstream SPEC? Is it only in the Fedora dist-git?

Yes, I'll change it in dist-git

File: tools/livecd-iso-to-pxeboot.sh (hunk 3 above)
Comment (lzap) [inline tools/livecd-iso-to-pxeboot.sh]: Sure, fixing.

**Comment by lzap** [comment]
Ok sorry for the delay, pushed the `/var/tmp` change.

**Comment by bcl** [comment]
pushed, and mkisofs is already used so no new requires. Thanks!

**Pull request closed without merging**